packages/neos-ui-editors/src/Editors/Button/index.js

```diff
@@ -0,0 +1,157 @@
+            })),
+
+            dataSourceIdentifier: PropTypes.string,
+            dataSourceUri: PropTypes.string,
+            dataSourceDisableCaching: PropTypes.bool,
+            dataSourceAdditionalData: PropTypes.objectOf(PropTypes.any)
+        }).isRequired,
+
+        i18nRegistry: PropTypes.object.isRequired
+    };
+
+    static defaultOptions = {
+        allowEmpty: false,
+        multiple: false,
+        disabled: false
+    };
+
+    state = {
+        active: [],
+        isLoading: false,
+        buttons: []
+    };
+
+    initialValueType = 'string';
+
+    constructor(props) {
+        super(props);
+
+        const {commit, options, value} = props;
+        this.initialValueType = Array.isArray(value) ? 'array' : 'string';
+        if (options.multiple && this.initialValueType === 'string') {
+            console.warn(`Misconfiguration in property "${props.identifier}". Multiple is activated but value type seems to be "string" but should be "array".`);
+        }
+
+        this.state = {
+            active: Array.isArray(value) ? value : (value ? [value] : []),
+            buttons: this.hasDataSource() ? [] : this.flattenValues(options.values)
+        };
+        this.commit = commit;
+    }
+
+    hasDataSource() {
+        return this.props.options.dataSourceIdentifier || this.props.options.dataSourceUri;
+    }
+
+    componentDidMount() {
+        if (this.hasDataSource()) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions() {
+        this.setState({isLoading: true});
+        this.props.dataSourcesDataLoader.resolveValue(getDataLoaderOptionsForProps(this.props), this.props.value)
+            .then(buttons => {
+                this.setState({
+                    isLoading: false,
+                    buttons
+                });
+            });
+    }
+
+    flattenValues(values) {
+        return Object.entries(values).map(([key, val]) => {
+            return {...val, value: key, label: this.props.i18nRegistry.translate(val.label)};
+        });
+    }
+
+    toggleActive(toggleValue) {
+        const {multiple, allowEmpty} = this.props.options;
+        // we'd like to have a copy instead of ref
+        let active = [...this.state.active];
+        if (multiple === false) {
+            if (active.includes(toggleValue)) {
+                active = [];
+            } else {
+                active = [toggleValue];
+            }
+        } else if (active.includes(toggleValue)) {
+            active.splice(active.indexOf(toggleValue), 1);
+        } else {
+            active.push(toggleValue);
+        }
+
+        // if allowEmpty is false but new active length will be 0, drop changes
+        if (!allowEmpty && active.length === 0) {
+            return;
+        }
+        this.setState({active});
+
+        this.commit(this.initialValueType === 'string' ? active[0] || '' : active);
+    }
+
+    isActive(val) {
+        return this.state.active.includes(val);
+    }
```

Comment:
```suggestion
    isSelected(value) {
        return this.state.selected.includes(value);
    }
```

For clarity:
The button editor shares select box semantics, so imho a value is not `active`, but `selected`.